openlcs/packages/mixins.py

```diff
@@ -210,3 +254,93 @@ def save_scan_result(self, **kwargs):
                     self.save_file_copyright_scan(list(set(file_ids)))
                     self.save_copyright_detections(path_file_dict, data)
                     self.update_scan_flag(source, "copyright_scan")
+
+
+class SaveContainerComponentsMixin:
+    def __init__(self):
+        self.components = None
+        self.release = None
+
+    @staticmethod
+    def create_component(component_data):
+        component, _ = Component.objects.update_or_create(
+            uuid=component_data.get("uuid"),
+            type=component_data.get("type"),
+            name=component_data.get("name"),
+            version=component_data.get("version"),
+            release=component_data.get("release"),
+            arch=component_data.get("arch"),
+            defaults={
+                "summary_license": component_data.get("license"),
+            },
+        )
+        return component
+
+    def build_release_node(self):
```

Comment:
I think functions in this mixins.py works, but it's probably more straightforward to move it to models.py using model method.

Anyway it's fine to keep it as it is.

Reply:
the product and release created in `create_product_release` when import